packages/lib/gpcircuits/circuits/proto-pod-gpc.circom

```diff
@@ -188,29 +197,69 @@ template ProtoPODGPC (
     }
 
     /*
-     * 1 OwnerModule with its inputs & outputs.
+     * External nullifier for owner modules (if any).
+     *
+     * The final nullifier hash(es) will be calculated based on this
+     * and the owner's identity.
+     */
+    signal input ownerExternalNullifier;
+    
+    /*
+     * <=1 OwnerModule with its inputs & outputs.
      */
 
     // Entry containing owner's Semaphore V3 commitment (public), or -1 to disable ownership checking.
-    signal input ownerEntryIndex;
+    signal input ownerEntryIndex[INCLUDE_OWNERV3];
```

Comment:
What's the tradeoff of empty arrays vs. optional inputs here?  I think these could all be wrapped in an if statement, and the compiler could use optional values in the input type.

Reply:
Separately, I think all the variable names which are duplicated for V4 below should have V3 in them here.  It'll mean more changes in the compiler, but I think it's better not to cause confusion about which is the "default" owner module.